lib/tasks/reports.rake

```diff
@@ -0,0 +1,32 @@
+require 'csv'
+require 'date'
+
+namespace :reports do
+  desc 'creates a csv of all submissions for a grant'
+  task :grant_submissions, [:grant] => [:environment] do |t, args|
+    grant = Grant.where(name: args[:grant]).first
+    submissions = grant.submissions
+
+    csv_file = "#{Rails.root}/tmp/#{grant}_submissions_#{Time.now.strftime("%Y-%m-%d-%H-%M-%S")}.csv"
+    puts "Writing submissions to file: #{csv_file}"
+
+    column_headers = [ "applicant last name", "applicant first name", "applicant email", "project title"]
+    questions = []
+    submissions.first.responses.each { |response| questions << response.question.text }
```

Comment:
Not all submissions may have all responses. Getting questions from first submission is not ideal. Instead you can get all questions from grant.questions, since grant has many questions.

questions = grant.questions

Reply:
The questions don't come back in the correct order. I think you can see this in your comment below.

Reply:
what do you mean in correct order? you can order by created_date or updated_date? or sequences, if questions have any

Reply:
you'll need to order by section.display_order and question.display_order